### PR TITLE
First iteration of error reporting, show memory usage, data row deletion

### PR DESCRIPTION
**Pull request by Karakatiza666**
https://github.com/feldera/feldera/issues/532 (home page only)
https://github.com/feldera/feldera/issues/528 (backend inconsistently reports of memory usage)
https://github.com/feldera/feldera/issues/525 (backend error)

### CODE REVIEW

File: web-console/src/pages/_app.tsx [hunk 1]
```diff
@@ -59,9 +58,7 @@ if (themeConfig.routingLoader) {
 // provide the default query function to your app with defaultOptions
 const queryClient = new QueryClient({
   defaultOptions: {
-    queries: {
-      queryFn: defaultQueryFn
-    }
+    queries: {}
```
Comment (gz) [inline web-console/src/pages/_app.tsx:61]: do we still need `defaultOption` here if queries is empty? do we need this at all anymore?

File: web-console/src/lib/services/pipelineManagerQuery.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
+// Define a default query function that will receive the query key and decide
+// what API to call based on the query key.
+//
+//
+// This interplays with react-query in the following way:
+// -  The defaultQueryFn is installed in the QueryClientProvider in _app.tsx
```
Comment (gz) [inline web-console/src/lib/services/pipelineManagerQuery.ts:6]: looks like comments here are outdated

File: web-console/src/lib/compositions/streaming/inspection/useDeleteRows.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import { useMutation } from '@tanstack/react-query'
+
+type Args = [
+  pipelineId: string,
+  relation: string,
+  rows: Partial<Record<'insert' | 'delete', Record<string, unknown>>>[],
+  isArray?: boolean
+]
+
+export function useInsertDeleteRows() {
```
Comment (gz) [inline web-console/src/lib/compositions/streaming/inspection/useDeleteRows.ts:14]: why is it called InsertDeleteRows wouldn't `useDeleteRows` be a better name?

**Review by gz — CHANGES_REQUESTED** [review]

**Comment by gz** [comment]
- [ ] This needs a space here between number and 'Rows'
Screenshot from 2023-09-04 02-23-31

- [ ] The copy button overlaps with text here can we make it so it's not overlapping: 
Screenshot from 2023-09-04 02-26-52

- [ ] Looks like pipeline failures currently don't have a report button, just the SQL programs? Ideally we can report also pipeline failures I think: 
Screenshot from 2023-09-04 02-26-52

- [ ] When I clicked report on a failed to compile SQL program, this happened instead of being able to report: 
Screenshot from 2023-09-04 02-28-22

**Comment by ryzhyk** [comment]
> https://github.com/feldera/feldera/issues/528 (backend inconsistently reports of memory usage)

@Karakatiza666 , what exactly is the problem with memory usage reporting?

**Comment by Karakatiza666** [comment]
@ryzhyk In all my builds memory usage via rss_bytes is only reported in GreenTrip pipeline, everywhere else it's zero

File: web-console/src/lib/compositions/streaming/inspection/useDeleteRows.ts (hunk 3 above)
Comment (Karakatiza666) [reply web-console/src/lib/compositions/streaming/inspection/useDeleteRows.ts:15]: I might need to come up with a better name yet - this specific function allows both insertion and deletion, and I don't want to artificially split it into two different hooks

**Comment by ryzhyk** [comment]
> @ryzhyk In all my builds memory usage via rss_bytes is only reported in GreenTrip pipeline, everywhere else it's zero

Does this happen when the pipeline is running?

File: web-console/src/lib/components/home/Health.tsx [hunk 4]
```diff
@@ -41,38 +46,59 @@ const programErrors = (program: ProgramDescr) =>
...
                   <br />
                   {program.program_id}
                 </>
-              )
+              ),
+              report: {
+                Error: '```\n' + limitAndAppend(e.message, 1000, '\n...Rest of the error...') + '\n```',
+                SQL: () =>
+                  ProgramsService.getProgram(program.program_id, true).then(
+                    p => '```\n' + limitAndAppend(p.code, 7000, '\n...Rest of the code...') + '\n```'
```
Comment (gz) [inline web-console/src/lib/components/home/Health.tsx:58]: If `7000` is the github limit we should probably make it a constant

**Comment by gz** [comment]
So few more things:

- [ ] The overlapping of the button with text in the background still looks ugly. 
Screenshot from 2023-09-05 17-00-56
Maybe the way to fix is is to have a header / title which ensures the button can't overlap with any text?
image

- [ ] I encountered this it looks like a regression (probably not in this PR) but it should re-order the connectors if they overlap (it does it correctly for inputs so there never is an overlap but somehow fails to do it for output)
Screenshot from 2023-09-05 17-02-11

- [ ] There is a now a searchbar in the drawer tables. We can probably have a search bar if there are many connectors but if there is just a few maybe we shouldn't display the search bar?
Screenshot from 2023-09-05 17-03-11

**Comment by Karakatiza666** [comment]
If this is a regression then not from this PR - behavior is the same on latest main;
Don't see the point in complicating the code with hiding the searchbar, atleast until it takes up space from something else

**Review by gz — APPROVED** [review]

**Pull request merged**